include/zephyr/arch/arc/v2/linker.ld

```diff
@@ -142,12 +142,12 @@ SECTIONS {
 #include <snippets-rodata.ld>
 #include <zephyr/linker/kobject-rom.ld>
 
-#if defined(CONFIG_CPP) && !defined(CONFIG_CPP_STATIC_INIT_GNU) && defined(__MWDT_LINKER_CMD__)
+#if defined(CONFIG_CPP) && !defined(CONFIG_CONSTR_STATIC_INIT_GNU) && defined(__MWDT_LINKER_CMD__)
```

Comment:
You could clean up the the kconfig logic here by having the CONSTR_STATIC_INIT_GNU default set =y when CPP.  Likewise I'd bet the MWDT usage is visible in kconfig and could force this off.

Reply:
I think it would help if I understood what case this is trying to cover. It seems like MWDT supports some form of C++ constructors, but in a fashion which is not compatible with how GCC/Clang do constructors? In which case, perhaps the logic here should just get moved to Kconfig and define CONFIG_STATIC_INIT_MWDT when it's in use?

Reply:
Although, perhaps in this case we can simply unilaterally remove the check for GNU-style constructors as MWDT doesn't support them anyways. Change these tests to `#if defined(CONFIG_CPP) && defined(__MWDT_LINKER_CMD__)`

Reply:
As it should be harmless,
I think I prefer to leave it, and let the arc maintainers decided if they want to remove it.

Reply:
Ok, now I'm just confused about the MWDT stuff. It seems that it does support constructors, placing their addresses in .ctors sections? If so, it really isn't any different from the GNU constructor mechanism, so we can
safely use the same code for both toolchains?

Reply:
Answered in https://github.com/zephyrproject-rtos/zephyr/pull/74700#discussion_r1650155756